group1/communication/node/main.js

```diff
@@ -131,6 +197,85 @@ client.get(room_url,function(data,response){
   });
 });
 
+var userVisit={
+    'save':function (userId) {
+        console.log("saved "+userId);
+        console.log(userId);
+        userVisit_template={
+            data:{
+                'user':util.format('http://localhost:8000/api/users/%s/', userId),
+            },
+            headers: { 'Content-Type': 'application/json' }
+        };
+        client.post("http://localhost:8000/api/userVisit/", userVisit_template, function (data, response) {
+            console.log("posted");
+            console.log(data);
+        });
+    },
+    'update':function (id) {
+        userVisit_template={
+            data:{
+                'id':id,
+                //'user':userId
+            },
+            headers: { 'Content-Type': 'application/json' }
+        };
+        client.put('http://localhost:8000/api/userVisit/', userVisit_template, function(data,response) {
+
+        });
+
+    },
+
+};
+
+var indv_messages={
+    'save':function(text, send_user, indvroom_id){
+        console.log("text - "+text);
+        console.log("send_user - "+send_user);
+        console.log("indvroom_id - "+indvroom_id);
+        indvtext_template={
+               data:{
+                   'text':text,
+                   'send_user':util.format('http://localhost:8000/api/users/%s/', send_user),
+                   'indv_room':indvroom_id,
+               },
+               headers: { 'Content-Type': 'application/json' }
+           };
+           client.post("http://localhost:8000/api/indvmessages/", indvtext_template, function (data, response) {
+
+                    console.log('check for repeat text '+data.text);
+                    io.sockets.in(indvroom_id).emit('indv_msg', data);
+
+           });
+
+    },
+    'update':function (msg) {
+            message_template = {
+                        data: {
+                                'id': msg.id,
+                                'text': msg.text
+                        },
+                        headers: { 'Content-Type': 'application/json' }
+            };
+            client.put('http://localhost:8000/api/indvmessages/', message_template, function(data,response) {
+                        global_namespace.emit('editmsg', msg);
+                        console.log( util.format('(%s) Message %s editted to: "%s"', response.statusCode, msg.id, msg.text) );
```

Comment:
Is this only being used for debugging or does it have usefulness outside of debugging?  If not, it can probably be removed.  If so, the space between "(" and "util" can be removed.  The space at the end between ")" and ");" can be removed as well.  Typo on "editted".  It should be "edited".